### PR TITLE
Issue 9793 introduced integrity-checker for roles in order to equate role ids

### PR DESCRIPTION
**Pull request by agomez-dotcms**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/integritycheckers/IntegrityUtil.java [hunk 1]
```diff
@@ -74,28 +73,47 @@ private File generateDataToFixCSV(String outputPath, String endpointId, Integrit
...
+                	);
+            		break;
+            	default:
+            		sbSelectTempTable.append("select remote_inode, local_inode from ");
+                    break;
             }
-
+            sbSelectTempTable.append(resultsTable).append(" where endpoint_id = ?");
+            
+            Connection conn = DbConnectionFactory.getConnection();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/integritycheckers/IntegrityUtil.java]: you could double check if the connection would need to be close

File: dotCMS/src/main/java/com/dotcms/integritycheckers/IntegrityUtil.java [hunk 2]
```diff
@@ -320,14 +337,14 @@ public void dropTempTables(final String endpointId) throws DotDataException {
...
             for (IntegrityType integrityType : types) {
-                final String tempTableName = getTempTableName(endpointId, integrityType);
-
-                if (doesTableExist(tempTableName)) {
-                    dc.executeStatement("truncate table " + tempTableName);
-                    dc.executeStatement("drop table " + tempTableName);
-                }else if (DbConnectionFactory.isMySql()){
-                    dc.executeStatement("drop table if exists " + tempTableName);
-                }
+            	for(String tempTableName : integrityType.getIntegrityChecker().getTempTableNames(endpointId)) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/integritycheckers/IntegrityUtil.java]: I think this part could be useful to have some logging, even for an else if the 

```}else if (DbConnectionFactory.isMySql()){```

Is false

File: src/com/dotcms/integritycheckers/RoleIntegrityChecker.java [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+import java.util.Stack;
+import java.util.function.Consumer;
+
+/**
+ * Role integrity checker implementation
+ * 
+ */
+public class RoleIntegrityChecker extends AbstractIntegrityChecker {
+
+	private RoleAPI roleAPI;
```
Comment (jdotcms) [inline src/com/dotcms/integritycheckers/RoleIntegrityChecker.java]: set final to make it immutable.

File: src/com/dotcms/integritycheckers/RoleIntegrityChecker.java [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+                }
+            }
+        } catch (SQLException e) {
+            throw new DotDataException(e.getMessage(), e);
+        }
+    }
+
+
+    private static class CmsRole {
+    	private String roleId;
```
Comment (jdotcms) [inline src/com/dotcms/integritycheckers/RoleIntegrityChecker.java]: you could make them final

**Review by jdotcms — APPROVED** [review]
I just made some minor comments but in general looks good, great job

**Review by freddyucv — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**